optimum/habana/transformers/models/starcoder2/modeling_starcoder2.py

```diff
@@ -61,17 +61,19 @@
 
 class GaudiStarcoder2MLP(Starcoder2MLP):
     def pre_mlp_forward(self, x):
-        inputs = self.act_fn(self.gate_proj(x)) * self.up_proj(x)
-        output = self.down_proj(inputs)
+        x = self.c_fc(x)
+        x = self.act(x)
+        x = self.c_proj(x)
+        output = F.dropout(x, p=self.residual_dropout, training=self.training)
         return output
```

Comment:
```suggestion
        x= F.dropout(x, p=self.residual_dropout, training=self.training)
        return x
```

@skavulya why do you create a new memory here, is there a reason you do not overwrite x?

Reply:
Good catch. I will overwrite x similar to the original implementation

https://github.com/huggingface/transformers/blob/f51ac9e059a78049362803c1d606a2c6a8160ee4/src/transformers/models/starcoder2/modeling_starcoder2.py#L199